### PR TITLE
Fixes to Add fast check for image orientation (#3568)

### PR DESCRIPTION
**Pull request by smz**
#### Description

This fixes some issues from #3568:
- it get rids of the new (**yet-to-be-released!**) JImageHelper class by incorporating its code into the JImage class.
- it correct some comment blocks
#### Test instruction

Assuming you are using the Protostar template, modify the template's index.php by inserting the following code after line 27 (`$sitename = $app->get('sitename');`):

```
echo '<pre>';

// Define the file to use. 
// NOTE:   Make sure this file exists or use another path to an existing image file!
$file = JPATH_BASE . '/images/powered_by.png';

echo str_repeat('_', 60) . "\n\n";
echo 'File is: ' . print_r($file, true) . "\n";
echo str_repeat('_', 60) . "\n\n";

// Get instance of JImage.
$image = new JImage($file); // $image must be instance of JImage

$orientation = $image->getOrientation();
echo 'For the $image object orientation is: ' . print_r($orientation, true) . "\n";
echo(str_repeat('_', 60)) . "\n\n";

// Get image properties calling static method from JImage::getImageFileProperties().
$props = JImage::getImageFileProperties($file);

echo 'Properties from JImage::getImageFileProperties($file);:';
echo "\n\n";
print_r($props);
echo "\n";
echo (str_repeat('_', 60)) . "\n\n";

echo 'Check for every property separately:' . "\n\n";

// Check the returned data contains all of these properties.
foreach (array(
   'width',
   'height',
   'type',
   'attributes',
   'bits',
   'channels',
   'mime',
   'filesize',
   'orientation') as $p
)
{
    if (array_key_exists($p, $props))
    {
        echo "{$p} is: " . print_r($props->$p, true) . "\n";
    }
    else
    {
        echo print_r("\$props does not contain {$p}-data", true) . "\n";
    }
}

echo(str_repeat('_', 60)) . "\n";
echo '</pre>';
die;
```

Visit your test site. The expected output is:

```
____________________________________________________________

File is: _PATH_TO_YOUR_JPATH_BASE_HERE_/images/powered_by.png
____________________________________________________________

For the $image object orientation is: landscape
____________________________________________________________

Properties from JImage::getImageFileProperties($file);:

stdClass Object
(
    [width] => 150
    [height] => 45
    [type] => 3
    [attributes] => width="150" height="45"
    [bits] => 8
    [channels] => 
    [mime] => image/png
    [filesize] => 4048
    [orientation] => landscape
)
____________________________________________________________

Check for every property separately:

width is: 150
height is: 45
type is: 3
attributes is: width="150" height="45"
bits is: 8
channels is: 
mime is: image/png
filesize is: 4048
orientation is: landscape
____________________________________________________________
```

### CODE REVIEW

**Comment by smz** [comment]
@Bakual, @itbra 
- I had to introduce a new `getOrientationString()` function as we had two `getOrientation()` functions in two different classes doing two different things

- since #3568 extended the amount of information returned by `getImageFileProperties()` and added the new public `getOrientation()` function, shouldn't we modify test units accordingly?

**Comment by smz** [comment]
Should this be tested or is it OK to just *code review* it?

File: libraries/joomla/image/image.php [hunk 1]
```diff
@@ -124,32 +141,85 @@ public function __construct($source = null)
...
+		return null;
+	}
+
+	/**
+	 * Compare width and height integers to determine image orientation.
+	 *
+	 * @param   integer  $width   The width value to use for calculation
+	 * @param   integer  $height  The height value to use for calculation
+	 *
+	 * @return  mixed    Orientation string or null.
```
Comment (smz) [inline libraries/joomla/image/image.php]: I think there is no way for this function to return a `null`...  :stuck_out_tongue_winking_eye:

**Comment by Bakual** [comment]
I don't think there is a need for a new method name. Just put the code of the method into the existing getOrientation. Instead of checking the two passed in arguments, you check the class properties.
That's the difference between a helper where you pass the dimension in to a static method as arguments or you use a OOP class where the dimensions are properties of the object.

**Comment by smz** [comment]
The new private `getOrientationString()` function is used in `getImageFileProperties()` too, so I think there is a good point for it being a function...

**Comment by smz** [comment]
... and thanks for clarifying the role of helper classes: very clear now!!

**Comment by Bakual** [comment]
> The new private getOrientationString() function is used in getImageFileProperties() too, so I think there is a good point for it being a function...

True, didn't catch that one.

> and thanks for clarifying the role of helper classes: very clear now!!

At least it's how I understand it. :smile:

**Comment by zero-24** [comment]
@smz can you just put some testinstructions here? I guess we have only smal different than here: https://github.com/joomla/joomla-cms/pull/3568?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/7060">issues.joomla.org/joomla-cms/7060</a>.</sub>

**Comment by smz** [comment]
@zero-24: absolutely, same testing instructions (*and I must admit I didn't test myself...* :innocent: )

**Comment by smz** [comment]
I suggest to assign this PR a 3.4.2 milestone, even if not yet RTC: if we don't merge this before 3.4.2 release we are going to introduce a new class with which we'll must live till 4.0...

**Comment by smz** [comment]
I decided to test this and apparently there are several issues: please stand-by before testing... :unamused:

**Comment by smz** [comment]
OK, **ready for testing**: instructions in  #3568 were not adequate, so I fixed them and added to this PR description.

**Comment by smz** [comment]
... and for what is worth, @test OK by me, but... is it correct that for `/images/powered_by.png` we don't have "*channels*"?

**Comment by itbra** [comment]
@test successful

**Comment by jwaisner** [comment]
@test 
PR test sucessful<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/7060">issues.joomla.org/joomla-cms/7060</a>.</sub>

**Comment by zero-24** [comment]
RTC based on testing. Thanks :smile: <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/7060">issues.joomla.org/joomla-cms/7060</a>.</sub>

**Comment by Bakual** [comment]
Merged into `staging`. Thanks.

**Pull request closed without merging**